### PR TITLE
chore: add rollover test

### PR DESCRIPTION
**Pull request by tustvold**
This test fails on current main.

Some print statements show that the row gets lost somewhere within the ReorgPlanner - i.e. the RUB is created missing a row

### CODE REVIEW

**Review by e-dard — APPROVED** [review]

**Comment by alamb** [comment]
Filed https://github.com/influxdata/influxdb_iox/issues/2072 to track losing rows in compact

**Comment by e-dard** [comment]
@tustvold I think the issue isn't necessarily to do with rollover. I added a commit

**Comment by alamb** [comment]
Interestingly,  when I turned on trace debugging it appears that the deduplicate algorithm in this case has decided there are duplicates (1) and has stripped it out for some reason. 

Continuing to dig...

```
Jul 20 16:22:38.352 DEBUG server::db::chunk: Input Predicate to read_filter predicate=Predicate { table_names: None, field_columns: None, partition_key: None, range: None, exprs: [] }
Jul 20 16:22:38.353 TRACE query::provider::deduplicate::algo: No dupes num_rows=2
Jul 20 16:22:38.353 TRACE query::provider::deduplicate::algo: dupes num_dupes=1 num_rows=2
```

File: server/src/db.rs [hunk 1]
```diff
@@ -1598,6 +1598,26 @@ mod tests {
         catalog_chunk_size_bytes_metric_eq(&test_db.metric_registry, "read_buffer", 0).unwrap();
     }
 
+    #[tokio::test]
+    async fn test_write_and_query() {
+        let db = make_db().await.db;
+
+        write_lp(db.as_ref(), "cpu bar=1 10").await;
+        write_lp(db.as_ref(), "cpu bar=2 10").await;
```
Comment (alamb) [inline server/src/db.rs:1606]: so these two rows have the same timestamp (10), so I think the code is (correctly) deduplicating them.

It is somewhat embarrassing it took me so long to catch that...

**Comment by alamb** [comment]
I fixed the timestamp in https://github.com/influxdata/influxdb_iox/pull/2068/commits/4f3c0d139838561d334c45ffcb419ec077604593 and now the tests seems to pass 🤔

File: server/src/db.rs (hunk 1 above)
Comment (alamb) [reply server/src/db.rs:1606]: If you run these commands to cherry pick 4f3c0d1 into this branch, you can see that fixing this number to `20` makes the test pass:

```
git checkout add-rollover-test
git cherry-pick 4f3c0d1
cargo test -p server -- rollover
```

Comment (tustvold) [reply server/src/db.rs:1606]: Oh ffs, I forget line protocol pits tags in the same block as the measurement 🤦‍♂️

**Comment by tustvold** [comment]
Sorry for the PEBCAC, this can be closed.

**Pull request closed without merging**